landoapi/workers/base.py

```diff
@@ -0,0 +1,165 @@
+
+        Args:
+            ssh_private_key (str): A string representing the private SSH key file.
+        """
+        # Set all the correct environment variables
+        agent_process = subprocess.run(
+            ["ssh-agent", "-s"], capture_output=True, universal_newlines=True
+        )
+
+        # This pattern will match keys and values, and ignore everything after the
+        # semicolon. For example, the output of `agent_process` is of the form:
+        #     SSH_AUTH_SOCK=/tmp/ssh-c850kLXXOS5e/agent.120801; export SSH_AUTH_SOCK;
+        #     SSH_AGENT_PID=120802; export SSH_AGENT_PID;
+        #     echo Agent pid 120802;
+        pattern = re.compile("(.+)=([^;]*)")
+        for key, value in pattern.findall(agent_process.stdout):
+            logger.info(f"_setup_ssh: setting {key} to {value}")
+            os.environ[key] = value
+
+        # Add private SSH key to agent
+        # NOTE: ssh-add seems to output everything to stderr, including upon exit 0.
+        add_process = subprocess.run(
+            ["ssh-add", "-"],
+            input=ssh_private_key,
+            capture_output=True,
+            universal_newlines=True,
+        )
+        if add_process.returncode != 0:
+            raise Exception(add_process.stderr)
+        logger.info("Added private SSH key from environment.")
+
+    @property
+    def _paused(self) -> bool:
+        """Return the value of the pause configuration variable."""
+        # When the pause variable is True, the worker is temporarily paused. The worker
+        # resumes when the key is reset to False.
+        return ConfigurationVariable.get(self.PAUSE_KEY, False)
+
+    @property
+    def _running(self) -> bool:
+        """Return the value of the stop configuration variable."""
+        # When the stop variable is True, the worker will exit and will not restart,
+        # until the value is changed to False.
+        return not ConfigurationVariable.get(self.STOP_KEY, False)
+
+    def _setup(self):
+        """Perform various setup actions."""
+        if hasattr(self, "ssh_private_key"):
+            self._setup_ssh(self.ssh_private_key)
+
+    def _start(self, max_loops=None, *args, **kwargs):
+        """Run the main event loop."""
+        # NOTE: The worker will exit when max_loops is reached, or when the stop
+        # variable is changed to True.
+        loops = 0
+        while self._running:
+            if max_loops is not None and loops >= max_loops:
+                break
+            while self._paused:
+                # Wait a set number of seconds before checking paused variable again.
+                self.throttle(self.sleep_seconds)
+            self.loop(*args, **kwargs)
+            loops += 1
+
+        logger.info(f"{self} exited after {loops} loops.")
+
+    @property
+    def throttle_seconds(self) -> int:
+        """The duration to pause for when the worker is being throttled."""
+        return ConfigurationVariable.get(self.THROTTLE_KEY, 3)
```

Comment:
```suggestion
        return int(ConfigurationVariable.get(self.THROTTLE_KEY, 3))
```

Reply:
As mentioned elsewhere I'll fix the type issue here after this patch lands.

Reply:
The suggestion you made here is technically redundant, as I [linked to](https://github.com/mozilla-conduit/lando-api/blob/main/landoapi/models/configuration.py#L64) elsewhere, this exact conversion is already made when calling `get`, and this change would be only to satisfy the type-checker/IDE (which is, in my opinion not a good enough reason to make this change.) Thus we should refactor this to make it possible to satisfy the type checker without having redundant code, but that refactoring would require some discussion and is out of scope for this PR.